### PR TITLE
build: add _Generic selection test to autoconf

### LINKED ISSUE #102: Add automake test for C99 standards compliance
**Is your feature request related to a problem? Please describe.**
A recent gcc version is required, even if non-GNU compilers, in order to support modern C99 C syntax.

**Describe the solution you'd like**
A automake `configure` check to warn users that their compiler isn't supported. (Otherwise, the first indication of a problem will be a compiler error on the new C99 syntax)

Until then, documentation describing this requirement should be added to README.md

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by underwoo**
configure will fail with a note if the C compiler does not support the _Generic selection introduced in C11 - https://en.cppreference.com/w/c/language/generic

Fixes #102

### CODE REVIEW

**Comment by ngs333** [comment]
@underwoo 
Was this tested?

**Review by ngs333 — APPROVED** [review]
Will wait for the merge of upstream PR before testing and merging.

**Comment by ngs333** [comment]
@ceblanton , @underwoo 
I ran into the exact smame issue Chris shows above (https://github.com/NOAA-GFDL/FRE-NCtools/pull/146#pullrequestreview-899610340). I was using analysis.

**Comment by underwoo** [comment]
I just noticed I forgot to add the gx_c__generic.m4 file to the commit.  I'll get that added, and push.

**Comment by underwoo** [comment]
@ngs333 @ceblanton I pushed the `GC_C__GENERIC` m4 macro.  Please try again.

**Comment by ngs333** [comment]
I verified the failure and appropriate message for the case when the compiler is not compliant.

**Pull request merged**
